src/renderer/qssWidget/js/qssMouseWidget.js

```diff
@@ -0,0 +1,444 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            swapMouseButtons: {
+                type: "gpii.qssWidget.mouseWidgetToggle",
+                container: "{that}.dom.swapMouseButtons",
+                options: {
+                    model: {
+                        setting: "{mouse}.model.setting.settings.swapMouseButtons"
+                    }
+                }
+            },
+            easierDoubleClick: {
+                type: "gpii.qssWidget.mouseWidgetToggle",
+                container: "{that}.dom.easierDoubleClick",
+                options: {
+                    model: {
+                        setting: "{mouse}.model.setting.settings.easierDoubleClick"
+                    }
+                }
+            },
+            largerMousePointer: {
+                type: "gpii.qssWidget.mouseWidgetToggle",
+                container: "{that}.dom.largerMousePointer",
+                options: {
+                    model: {
+                        setting: "{mouse}.model.setting.settings.largerMousePointer"
+                    }
+                }
+            },
+            channelNotifier: {
+                type: "gpii.psp.channelNotifier",
+                options: {
+                    events: {
+                        // Add events the main process to be notified for
+                        onQssWidgetSettingAltered:       "{mouse}.events.onQssWidgetSettingAltered",
+                        onQssWidgetNotificationRequired: "{mouse}.events.onQssWidgetNotificationRequired"
+                    }
+                }
+            }
+        }
+    });
+
+    /**
+     * Either increases or decreases the current setting's value (depending on the
+     * `shouldSubtract` parameter) with the `divisibleBy` amount specified in the
+     * setting's schema. It also takes care that the new value of the setting does
+     * not become bigger/smaller than the maximum/minimum allowed value for the
+     * setting.
+     * @param {Component} that - The `gpii.qssWidget.mouseSpeedStepper` instance.
+     * @param {Number} value - The initial value of the setting before the operation.
+     * @param {Object} schema - Describes the schema of the setting.
+     * @param {Number} schema.min - The minimum possible value for the setting.
+     * @param {Number} schema.max - The maximum possible value for the setting.
+     * @param {Number} schema.divisibleBy - The amount which is added or subtracted
+     * from the setting's value every time this function is invoked.
+     * @param {Boolean} shouldSubtract - Whether the `divisibleBy` amount should be
+     * subtracted from or added to the setting's value.
+     * @return {Boolean} Whether there was a change in the setting's value.
+     */
+    gpii.qssWidget.mouse.makeRestrictedStep = function (that, value, schema, shouldSubtract) {
```

Comment:
Let's try and unify this with baseStepper.makeRestrictedStep. What is the nature of the edge case which gives rise to restrictedValue of 2 - can we fix this by simply adjusting the schema before passing it into this function?
Other irregularities
 - "shouldSubtract" can easily be represented using "stepMultiplier" scheme from the other function
 - why do we use a precision of 2 here and 3 in the other function?

Reply:
We not reusing the original one from the base stepper grade because the windows mouse speed setting cannot accept a value of 0 and the logic need to be slightly different. 
Also, we use a precision of 3 in the base stepper because we need that precision for the volume setting value.

Reply:
But as far as I can see, the value 0 is not within the valid schema range?
https://github.com/GPII/universal/blob/master/testData/solutions/win32.json5#L8184

Reply:
Yes, but because in this case, the stepper uses even multiplier. The setting min value is odd and broke the calculations.

Reply:
I think we could fold this into a general purpose algorithm which always guarantees that a restricted value is a multiple of the step - it would be very helpful to axe this bulky and largely duplicated function.

Reply:
Sooo, about that - the reason we are doing this is because of the problem with mouseSpeed not accepting 0 as a value (basically doing nothing). So if we are using the default logic (1 to 20, division by 2) then when launching the app and clicking "Faster" the values will be 1, 3, 5, and at the end 20 (because will hit the max value). When going back the value will be even 20, 18, 16 and eventually 1 (because will hit the low value). This by itself its not that horrible, but the indicators itself are created initially and they will follow the second logic (20, 18, 16...) so it will work only when hit the max value first.

I'm sure there is a possible way this to be solved by some math solution, but we don't have the time to do it right now. And with that we have 2 solutions - the first one is to leave it as it is; to change the schema from 2 to 20 (or from 1 to 19) just to make it even. So... pick your poison :smiley_cat:

Reply:
Don't worry - the poison will not be that bad :)
var multiplier = 1000;
var adjustedMin = Math.ceil(multiplier * schema.min/ schema.divisibleBy) / multiplier;
var adjustedMax = Math.floor(multiplier * schema.max / schema.divisibleBy) / multiplier;
You can then harmonise your parseFloat line to this scheme by adjusting it to 
value = Math.round(multiplier * (value - step)) / multiplier;
This rounding will not be perfect but should be good enough for the current purposes, and express the intention consistently.